src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -668,6 +668,11 @@ private static bool IsViableType(LookupResult result)
                         symbols.Add(bindingResult);
                         bindingResult = null;
                     }
+
+                    if (bindingResult?.Kind == SymbolKind.NamedType)
+                    {
+                        ConstraintsHelper.CheckTupleNamesConstraints((NamedTypeSymbol)bindingResult, node.Location, diagnostics, TupleNamesCheckKind.DeepExceptTopLevel);
```

Comment:
>ConstraintsHelper.CheckTupleNamesConstraints((NamedTypeSymbol)bindingResult, node.Location, diagnostics, TupleNamesCheckKind.DeepExceptTopLevel); [](start = 24, length = 145)

Consider adding a comment describing what would be a scenario that is getting caught by a check here.